vllm/entrypoints/openai/serving_chat.py

```diff
@@ -145,15 +157,39 @@ async def chat_completion_stream_generator(
                 if finish_reason_sent[i]:
                     continue
 
+                if request.echo and request.max_tokens == 0:
+                    delta_token_ids = res.prompt_token_ids
+                    top_logprobs = res.prompt_logprobs
+                elif request.echo and request.max_tokens > 0:
+                    delta_token_ids = res.prompt_token_ids + output.token_ids
+                    top_logprobs = res.prompt_logprobs + (output.logprobs
+                                                          or [])
+                else:
+                    delta_token_ids = output.token_ids[previous_num_tokens[i]:]
+                    top_logprobs = output.logprobs[
+                        previous_num_tokens[i]:] if output.logprobs else None
+
+                if request.logprobs is not None:
+                    assert(top_logprobs is not None),\
+                           "top_logprobs must be provided when logprobs is requested"
+                    logprobs = create_logprobs_fn(
```

Comment:
Why not invoke `self._create_logprobs` directly?

Reply:
This is what `serving_completion.py` did. I can only speculate on their design decision but perhaps it was to allow for extensibility with other log probability formats in the future--not just the open ai ones?

Reply:
I am happy to change it if you think that is best.

Reply:
Because this function is already defined in a class, so there's no need to delegate a callable function here. Finally we can delete this and relevant things and invoke parent method `_create_logprobs` directly.